docs/docs/building-modules/14-depinject.md

```diff
@@ -0,0 +1,120 @@
+---
+sidebar_position: 1
+---
+
+# Dependency Injection
+
+:::note
+
+### Pre-requisite Readings
+
+* [Cosmos SDK Dependency Injection Framework](../tooling/02-depinject.md)
+
+:::
+
+[`depinject`](../tooling/02-depinject.md) is used to wire any module in `app.go`.
+All core modules are already configured to support dependency injection.
+
+To work with `depinject` a module must define its configuration and requirements so that `depinject` can provide the right dependencies.
+
+In brief, as a module developer, the following steps are required:
+
+1. Define the module configuration using Protobuf
+2. Define the module dependencies in `x/{moduleName}/module.go`
+
+A chain developer can then use the module by following these two steps:
+
+1. Configure the module in `app_config.go`
+2. Inject the module in `app.go`
+
+## Module Configuration
+
+The module available configuration is defined in a Protobuf file, located at `{moduleName}/module/v1/module.proto`.
+
+```protobuf reference
+https://github.com/cosmos/cosmos-sdk/blob/main/proto/cosmos/group/module/v1/module.proto
+```
+
+* `go_import` must point to the Go package of the custom module.
+* Message fields define the module configuration.
+  That configuration can be set in the `app_config.go` / `app.yaml` file for a chain developer to configure the module.  
+  Taking `group` as example, a chain developer is able to decide, thanks to `uint64 max_metadata_len`, what the maximum metatada length allowed for a group porposal is.
+
+  ```go reference
+    https://github.com/cosmos/cosmos-sdk/blob/0d8787c/simapp/app_config.go#L202-L206
+  ```
+
+That message is generated using [`pulsar`](https://github.com/cosmos/cosmos-sdk/blob/main/scripts/protocgen-pulsar.sh) (by running `make proto-gen`).
```

Comment:
Pulsar is totally optional. They can use the official `protoc-gen-go` if they like: https://developers.google.com/protocol-buffers/docs/reference/go-generated

Reply:
Clarified!